ql/lib/codeql/kaleidoscope/dataflow/Ssa.qll

```diff
@@ -0,0 +1,203 @@
+private import codeql.ssa.Ssa as SsaLib
+private import codeql.kaleidoscope.controlflow.BasicBlocks
+private import codeql.Locations
+private import codeql.kaleidoscope.Cfg as Cfg
+private import codeql.kaleidoscope.Ast
+
+private class Assignment extends AstNode {
+  private Identifier variable;
+
+  Assignment() {
+    this =
+      any(BinaryOpExpression e |
+        e.getOperator().getName() = "=" and
+        variable = e.getLhs().(VariableExpression).getVariable()
+      ) or
+    variable = this.(Initializer).getVariable()
```

Comment:
Perhaps have another field `rhs`, to avoid duplicating this pattern in `getValue`.

Reply:
The `rhs` is optional  in a variable declaration (Initializer) in which case it defaults to `0.0`.